packages/studioCMSBlog/src/pages/blog/[slug].astro

```diff
@@ -0,0 +1,46 @@
+---
+import { BlogLayout } from "studiocms-blog:layouts";
+import { PostHeader } from "studiocms-blog:components";
+import { StudioCMSRenderer, contentHelper } from "studiocms:components";
+
+// Get the slug from the URL
+const { slug } = Astro.params;
```

Comment:
The file `[slug].astro` could use a rest parameter like `[...slug].astro` to create nested paths/slugs